utilities/git/hooks/pre-commit.sh

```diff
@@ -14,7 +14,10 @@ then
    exit 1
 fi
 
-clang-format \
-    -i \
-    -style=file \
-    $(git diff --cached --name-only *.cpp *.h)
+for file in $(git diff-index --cached --name-only HEAD | egrep ".*(\.cpp|\.h)$") ;
```

Comment:
`$(git diff-index --cached --name-only HEAD | egrep ".*(\.cpp|\.h)$")` should have the same behaviour. Globbing could be an issue, but could be avoided with `"*.cpp" "*.h"`.

That still leaves the problem that neither this, nor the original, actually add the modified files to the commit. They also modify the current file rather than the staged file. I think we need to identify the cached cpp/h files, store the current, possibly modified version, format the cached version, git add the formatted version, recover the current, possibly modified version, then format that too so it matches.

Maybe something like:
```
#Change the delimiter used by 'for' so we can handle spaces in names
IFS=$'\n'
for FILE in $(git diff --cached --name-only "*.cpp" "*.h")
do
  # Store the current version in case it has been modified
  TMPFILE="$(mktemp)"
  cp "$FILE" "$TMPFILE"
  # Temporarily remove modified versions, if they exist
  git checkout "$FILE"
  # Format
  clang-format -i "$FILE"
  # Add the formatted file
  git add "$FILE"
  # Restore the old, possibly modified version
  mv "$TMPFILE" "$FILE"
  # Format the uncached version to keep it in sync with the cached
  clang-format -i "$FILE"
done

# Don't commit if there are no changes to commit
test "$(git diff --cached)" != ""

```